### ISSUE TITLE
FEAT Habits

### ISSUE THREAD

**Issue by richardfeynmanrocks** [Shabang-Systems/Condution#127]
**Is your feature request related to a problem? Please describe.**
The addition of a new specialized task — the *habit* — a la Org-Mode's [org-habit](https://orgmode.org/manual/Tracking-your-habits.html). 

Originally if one wanted to add a habitual task like "Water the Plants" you would create a repeating task that you would check off each day, and if you didn't do it it would be marked as overdue the next day. 

However, this doesn't really make sense. Habits should be optional and not have "due dates" but a metric of how often you want to do them. For bonus points, features like org-habit's consistency grap, habit log, and habit-view-calendar-thing would be cool too.

**Describe the solution you'd like**
Add a new variant of tasks with these properties.

**Additional context**
This one is a little out there but I think it would expand this app's functionality.

**Comment by Exr0n** [comment]
I guess we should group this with #83... I'll think about what the best way to implement this would be.

**Comment by Jemoka** [comment]
Not really. #83 is about better repeating. This is about basically... optional tasks.

**Comment by Cosmic-Goat** [comment]
[Habitica](https://habitica.com) is an interesting gamification take on the habit-task concept, as a prospective Condution user I'd love to see a potential integration with their [API](https://habitica.fandom.com/wiki/Application_Programming_Interface) to handle this functionality :)

**Comment by Jemoka** [comment]
We are building out a full-fleshed API for Condution that’s capable of JS, C++, and python bindings. Stay tuned! More info to come after Christmas.

thanks for checking out condution!

**Comment by richardfeynmanrocks** [comment]
Still would like much of the features described in [org-habit](https://orgmode.org/manual/Tracking-your-habits.html) to have an intuitive counterpart in Condution! Expanding the featurebase beyond what other existing task managers like OmniFocus could really make Condution different.

**Comment by Jemoka** [comment]
The best solution is probably going to be finishing the Christmas dev-wasm sprint, and creating a condution.el backend that plugs in as a db provider.
W.r.t. habituca, that could be a hook on the JS client API

Sincerely yours,
Houjun "Jack" Liu

**Comment by jsmollin** [comment]
> The best solution is probably going to be finishing the Christmas dev-wasm sprint, and creating a condution.el backend that plugs in as a db provider.
> W.r.t. habituca, that could be a hook on the JS client API
> 
> Sincerely yours,
> Houjun "Jack" Liu
> […](#)

Did you finish that sprint?
